### PR TITLE
Allow valid contextual keyword completions even when an auto-import by the same name exists

### LINKED ISSUE #55607: Missing `type` keyword completion causes auto-import of `type` symbol
With `@types/node` installed, start typing `type` at the top level in any TS file 😱 

image

### PR DESCRIPTION
**Pull request by andrewbranch**
See code comment and linked issue comments for description.

Fixes #55607

### CODE REVIEW

File: src/services/completions.ts [hunk 1]
```diff
@@ -5829,3 +5833,31 @@ function toUpperCharCode(charCode: number) {
...
+ * may be the beginning of a type alias declaration (keyword completion), or
+ * it may be the beginning of
+ *
+ * ```ts
+ * import { type } from "os";
+ * type() === "Darwin" ? doSomething() : doSomethingElse();
+ * ```
+ */
+function isContextualKeywordInAutoImportableExpressionSpace(keyword: string) {
+    return keyword === "abstract" ||
```
Comment (jakebailey) [inline src/services/completions.ts:5856]: I think canonically we'd use a switch for this (faster at runtime I assume).

**Review by gabritto — APPROVED** [review]

File: src/services/completions.ts (hunk 1 above)
Comment (andrewbranch) [reply src/services/completions.ts:5856]: Didn’t we move the visitor API off a `switch` and onto a lookup table for performance reasons?

Comment (andrewbranch) [reply src/services/completions.ts:5856]: Anyway, this list seems small enough and is unlikely to grow

Comment (jakebailey) [reply src/services/completions.ts:5856]: No, that was for a different reason IIRC, the function was so big that it didn't optimize with so many cases handled inside of it

**Pull request merged**